### PR TITLE
[JSON-RPC] add API call metrics

### PR DESCRIPTION
**Pull request by 0xcb9ff9**
# Description

json-rpc add call count metrics

# Changes include

- [x] New feature (non-breaking change that adds functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels

## Testing

- [x] I have tested this code with the official test suite

### CODE REVIEW

File: jsonrpc/metrics.go [hunk 1]
```diff
@@ -17,6 +17,51 @@ type Metrics struct {
...
+	// TxPool metrics
+	txPoolAPI *stdprometheus.CounterVec
+
+	// Debug metrics
+	debugAPI *stdprometheus.CounterVec
+}
+
+func (m *Metrics) EthAPICounterInc(method string) {
+	if m.ethAPI != nil {
+		m.ethAPI.With(stdprometheus.Labels{"method": "eth_" + method}).Inc()
```
Comment (DarianShawn) [inline jsonrpc/metrics.go]: why not use constant method names instead of runtime concat?

**Review by DarianShawn — COMMENTED** [review]
Only a few suggestions to go.

File: jsonrpc/metrics.go (hunk 1 above)
Comment (0xcb9ff9) [reply jsonrpc/metrics.go]: @DarianShawn now use type parameters

**Review by DarianShawn — APPROVED** [review]
Thanks for your contribution to observability.
LGTM

**Pull request merged**